### PR TITLE
render callback + multiple clusters + infowindow fix on ViaMich

### PR DESCRIPTION
**Pull request by kavrillon**

### CODE REVIEW

File: src/providers/viaMichelin/Marker.js [hunk 1]
```diff
@@ -35,7 +25,19 @@ class Marker {
         }));
 
         marker.id = point.id;
+        
+        if (options && options.activeInfoWindow) {
+            if (typeof options.htm === 'function') {
 
+                marker.addEventListener("onClick", function() {
```
Comment (dpellier) [inline src/providers/viaMichelin/Marker.js]: `() =>`

File: src/providers/viaMichelin/ViaMichelin.js [hunk 2]
```diff
@@ -133,48 +139,56 @@ class ViaMichelinMap extends Map {
...
 
         let markers = [];
         let options = {};
-        let activeInfoWindow;
 
         for(let i = 0; i < points.length; i++) {
             options = Object.assign({}, this.options.marker, points[i].options ? points[i].options : {});
-            activeInfoWindow = points[i].activeInfoWindow !== undefined ? points[i].activeInfoWindow : this.options.activeInfoWindow;
+
+            if (options.activeInfoWindow === undefined) {
```
Comment (dpellier) [inline src/providers/viaMichelin/ViaMichelin.js]: this will allow `null`

File: src/providers/viaMichelin/ViaMichelin.js [hunk 3]
```diff
@@ -133,48 +139,56 @@ class ViaMichelinMap extends Map {
...
+                options.activeInfoWindow = this.options.activeInfoWindow;
+            }
+
+            if (options.activeCluster === undefined) {
```
Comment (dpellier) [inline src/providers/viaMichelin/ViaMichelin.js]: this will allow `null`

File: src/providers/viaMichelin/ViaMichelin.js [hunk 4]
```diff
@@ -133,48 +139,56 @@ class ViaMichelinMap extends Map {
...
-                // Reset all markers & clusters because ViaMichelin API does not allow it...
-                this.map.removeAllLayers();
-                this.cluster.clear();
-
-                // Redraw each markers
-                this.markers.forEach((marker) => {
-                    this.map.addLayer(marker);
-                });
+    setCluster(markers, clusterIndex, clusterConfig) {
+        if (this.map && this.options.activeCluster && clusterIndex !== false) {
```
Comment (dpellier) [inline src/providers/viaMichelin/ViaMichelin.js]: clusterIndex should be a `number`  ?

File: src/providers/googleMap/googleMap.js [hunk 5]
```diff
@@ -185,36 +176,67 @@ class GoogleMap extends Map {
...
+
+        let markers = [];
         let options = {};
-        let activeInfoWindow;
 
         for(let i = 0; i < points.length; i++) {
             options = Object.assign({}, this.options.marker, points[i].options ? points[i].options : {});
-            activeInfoWindow = points[i].activeInfoWindow !== undefined ? points[i].activeInfoWindow : this.options.activeInfoWindow;
+
+            if (options.activeInfoWindow === undefined) {
```
Comment (dpellier) [inline src/providers/googleMap/googleMap.js]: this will allow `null`

File: src/providers/googleMap/googleMap.js [hunk 6]
```diff
@@ -185,36 +176,67 @@ class GoogleMap extends Map {
...
+                options.activeInfoWindow = this.options.activeInfoWindow;
+            }
+
+            if (options.activeCluster === undefined) {
```
Comment (dpellier) [inline src/providers/googleMap/googleMap.js]: this will allow `null`

File: src/providers/googleMap/googleMap.js [hunk 7]
```diff
@@ -185,36 +176,67 @@ class GoogleMap extends Map {
...
                 });
             }
 
+            markers.push(marker);
             this.markers.push(marker);
+        }
 
-            if (this.map && this.plugins.clusterer && this.options.activeCluster) {
-                this.markerClusterer.addMarker(marker);
+        // If clustering is activated for those poses
```
Comment (dpellier) [inline src/providers/googleMap/googleMap.js]: `those markers`

File: src/providers/googleMap/googleMap.js [hunk 8]
```diff
@@ -185,36 +176,67 @@ class GoogleMap extends Map {
...
+        if (this.map && this.plugins.clusterer && this.options.activeCluster && clusterIndex !== false) {
```
Comment (dpellier) [inline src/providers/googleMap/googleMap.js]: clusterIndex => `number` ?

File: src/providers/viaMichelin/ViaMichelin.js (hunk 4 above)
Comment (kavrillon) [inline src/providers/viaMichelin/ViaMichelin.js]: in fact it's the index of `this.markerClusterers`, or false if you don't want to clusterize your markers.
Maybe I should use `null` instead...

Comment (dpellier) [inline src/providers/viaMichelin/ViaMichelin.js]: yep

**Pull request merged**